rsfbclient-rust/src/blr.rs

```diff
@@ -59,17 +59,18 @@ pub fn params_to_blr(
     for p in params {
         match p {
             Param::Text(s) => {
-                if s.len() > MAX_DATA_LENGTH {
+                let bytes = conn.charset.encode(s.to_string())?;
```

Comment:
If I'm not wrong, here the call `s.to_string()` makes a copy of the content of variable `s` and `charset.encode()` does a second copy, right?

This could be replaced by something like:
``` rust
    for p in params {
        match p {
            Param::Text(&'a s) => {
                if s.len() > MAX_DATA_LENGTH {
                let bytes = conn.charset.encode<'a>(&s)?;
```

I have the impression that code in this place will be very hot and optimizing or helping the compiler optimize will contribute for raw performance.

Reply:
In this case, just removing the `to_string()` call will fix the unnecessary copy.

We should not change the encode to take a reference, because in the present case, we pass a owned string,
and if the encoding is utf8, we can just return the passed string underlying vector, without making any copy,
however, when the encoding is not utf8, we need to make a copy, as the bytes will likely be different than the ones
in the original string.

Reply:
I didn't understand exactly if you mean:

 - [x] remove `to_string()`
- [ ] change `charset.encode()` signature

Reply:
This:
```suggestion
                let bytes = conn.charset.encode(s)?;
```

Reply:
Good catch. Fixed!

This 'to_string()' remained of an old encode method version.